app/models/ems_refresh/save_inventory_physical_infra.rb

```diff
@@ -123,11 +123,19 @@ def save_physical_disk_inventory(parent, hash)
   end
 
   #
-  # Saves the canister information of a storage
+  # Saves the canisters information of a storage
   #
-  def save_canister_inventory(physical_storage, hash)
-    return if hash.nil?
-    save_inventory_single(:canister, physical_storage, hash)
+  def save_canisters_inventory(physical_storage, hashes, target = nil)
+    return if hashes.nil?
+    deletes = target == physical_storage ? :use_association : []
```

Comment:
Same comment as the physical_disks one, this should just be `:use_association`

Reply:
@agrare Also changed it.